server/repository/src/migrations/v1_01_12/barcode.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+            CREATE TABLE barcode (
+                id text NOT NULL PRIMARY KEY,
+                value text NOT NULL,
+                item_id text REFERENCES item(id),
+                manufacturer_id text,
+                pack_size int4,
+                parent_id text
```

Comment:
Some naive questions: why do we need the barcode table? Isn't the barcode just a number? And shouldn't the item include this number together with the other values like `pack_size` and `manufacture_id`? Or could there be multiple barcodes for the same item? or could one item have multiple pack_sizes? Would have thought the manufacture has exactly one barcode for one product?

What is `parent_id`?

Reply:
> why do we need the barcode table? 

Yes, an item can have multiple pack size. Pack size is stored for a specific stock line and not at an item level

> What is parent_id?

the `parent_id` is in mSupply. The [docs say](https://docs.msupply.org.nz/barcode_scanning:adding_barcodes_to_items)  "Parent: the number (shown in the Number column) of the barcode that is the parent (i.e. has the next largest pack size) of this barcode"

I've implemented in case its helpful, even though I have no intention of using it right now. That's a poor justification, I know.